src/main/scala/com/typesafe/sbt/packager/archetypes/scripts/CommonStartScriptGenerator.scala

```diff
@@ -0,0 +1,137 @@
+  }
+
+  protected[this] type SpecializedScriptConfig <: ScriptConfig
+
+  protected[this] def generateStartScripts(config: SpecializedScriptConfig,
+                                           mainClass: Option[String],
+                                           discoveredMainClasses: Seq[String],
+                                           targetDir: File,
+                                           log: sbt.Logger): Seq[(File, String)] =
+    StartScriptMainClassConfig.from(mainClass, discoveredMainClasses) match {
+      case NoMain =>
+        log.warn("You have no main class in your project. No start script will be generated.")
+        Seq.empty
+      case SingleMain(main) =>
+        Seq(createMainScript(main, config, targetDir, Seq(main)))
+      case MultipleMains(mains) =>
+        generateMainScripts(mains, config, targetDir, log)
+      case ExplicitMainWithAdditional(main, additional) =>
+        createMainScript(main, config, targetDir, discoveredMainClasses) +:
+          createForwarderScripts(config.executableScriptName, additional, targetDir, config, log)
+    }
+
+  private[this] def generateMainScripts(discoveredMainClasses: Seq[String],
+                                        config: SpecializedScriptConfig,
+                                        targetDir: File,
+                                        log: sbt.Logger): Seq[(File, String)] = {
+    val classAndScriptNames = ScriptUtils.createScriptNames(discoveredMainClasses)
+    ScriptUtils.warnOnScriptNameCollision(classAndScriptNames, log)
+    classAndScriptNames.map {
+      case (qualifiedClassName, scriptName) =>
+        val newConfig = config.withScriptName(scriptName)
+        createMainScript(qualifiedClassName, newConfig, targetDir, discoveredMainClasses)
+    }
+  }
+
+  private[this] def mainScriptName(config: ScriptConfig): String =
+    config.executableScriptName + scriptSuffix
+
+  /**
+    *
+    * @param mainClass - Main class added to the java command
+    * @param config - Config data for this script
+    * @param targetDir - Target directory for this script
+    * @return File pointing to the created main script
+    */
+  private[this] def createMainScript(mainClass: String,
+                                     config: SpecializedScriptConfig,
+                                     targetDir: File,
+                                     mainClasses: Seq[String]): (File, String) = {
+    val template = resolveTemplate(config.templateLocation)
+    val replacements = createReplacementsForMainScript(mainClass, mainClasses, config)
+    val scriptContent = TemplateWriter.generateScript(template, replacements, eol, keySurround)
+    val scriptNameWithSuffix = mainScriptName(config)
+    val script = targetDir / scriptTargetFolder / scriptNameWithSuffix
+    IO.write(script, scriptContent)
+    // TODO - Better control over this!
+    if (makeScriptsExecutable)
+      script.setExecutable(true)
```

Comment:
You you simply do
```scala
script.setExecutable(makeScriptsExecutable)
```

Reply:
Yes, assuming `script.setExecutable(false)` is no-op and works well on Windows :) Meanwhile, does building zips with executable bits set on Windows works at all?

Reply:
Of course, I don't hesitate to simplify this line, if it is possible.

Reply:
Good point. No idea what happens on windows. We can leave this as is and should put a comment in there that `setExecutable` may behave differently on different platforms.